caddy.go

```diff
@@ -405,13 +410,10 @@ func startWithListenerFds(cdyfile Input, inst *Instance, restartFds map[string]r
 		return err
 	}
 
-	if restartFds == nil {
-		// run startup callbacks since this is not a restart
-		for _, startupFunc := range inst.onStartup {
-			err := startupFunc()
-			if err != nil {
-				return err
-			}
+	for _, startupFunc := range inst.onStartup {
+		err := startupFunc()
+		if err != nil {
+			return err
```

Comment:
Yes, otherwise the `log` and `errors` directives will write to nil and panic on each request since they open the log file in the onStartup call.
Or we can fix that in `log` or `errors` directives instead, what do you think?